test.py

```diff
@@ -202,11 +210,60 @@ def test_sample_without_suffix(self):
     def tearDownClass(cls):
         os.unlink(cls.test_db)
 
+class TestRules(unittest.TestCase):
+    """
+    Unittests for Rules.
+
+    @author: Felix Bauer
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.conf = PeekabooDummyConfig()
+        cls.factory = SampleFactory(cuckoo = None,
+                db_con = None,
+                connection_map = None,
+                base_dir = cls.conf.sample_base_dir,
+                job_hash_regex = cls.conf.job_hash_regex,
+                keep_mail_data = False)
+        cls.sample = cls.factory.make_sample(os.path.realpath(__file__))
+
+    def test_rule_file_type_on_whitelist(self):
+        combinations = [
+            [False, ['text/plain']],
+            [True, ['application/vnd.ms-excel']],
+            [True, ['text/plain', 'application/vnd.ms-excel']],
+            [True, ['image/png', 'application/zip', 'application/vnd.ms-excel']],
+            [True, ['', 'asdfjkl', '93219843298']],
+            [False, []], # should this really result in no further_analysis?
```

Comment:
No, IMO us not knowing any MIME type should not mean that the sample is whitelisted. In python's set logic however, the empty set obviously is a subset of any set. So we'd need an additional `len(s.mimetypes) > 0 and` in `file_type_on_whitelist` in `rules.py`. Would you like to add it to this PR as a followup commit or should I open an issue and do my own PR?

Reply:
Just realised: It's the same for greylisting: There we use set intersection and the intersection of the empty set with any set is empty and causes the greylist not to apply if we do not know any MIME type.

Reply:
Looks good.